raiden/blockchain/state.py

```diff
@@ -69,72 +71,54 @@ class NewChannelDetails:
         chain_state=chain_state, canonical_identifier=canonical_identifier
     )
 
-    # This may happen for two reasons:
-    # - This node is not a participant for the given channel (normal operation,
-    #   the event should be ignored).
-    # - Something went wrong in our code and the channel state was cleared
-    #   before settle (a bug, this should raise an exception on development
-    #   mode).
-    # Because we cannot distinguish the two cases, assume the channel is not of
-    # interest and ignore the event.
     if not channel_state:
         return None
 
-    # Recover the locksroot from the blockchain to fix data races. Check
-    # get_onchain_locksroots for details.
-    try:
-        # First try to query the unblinded state. This way the
-        # ContractReceiveChannelSettled's locksroots will  match the values
-        # provided during settle.
-        our_locksroot, partner_locksroot = get_onchain_locksroots(
-            proxy_manager=proxy_manager,
-            channel_state=channel_state,
-            participant1=channel_state.our_state.address,
-            participant2=channel_state.partner_state.address,
-            block_identifier=block_hash,
-        )
-    except ValueError:
-        # State pruning handling. The block which generate the
-        # ChannelSettled event may have been pruned, because of this the
-        # RPC call raised ValueError.
-        #
-        # The solution is to query the channel's state from the latest
-        # *confirmed* block, this /may/ create a ContractReceiveChannelSettled
-        # with the wrong locksroot (i.e. not the locksroot used during the call
-        # to settle). However this is fine, because at this point the channel
-        # is settled, it is known that the locksroot can not be reverted
-        # without an unlock, and because the unlocks are fair it doesn't matter
-        # who called it, only if there are tokens locked in the settled
-        # channel.
-        our_locksroot, partner_locksroot = get_onchain_locksroots(
-            proxy_manager=proxy_manager,
-            channel_state=channel_state,
-            participant1=channel_state.our_state.address,
-            participant2=channel_state.partner_state.address,
-            block_identifier=current_confirmed_head,
-        )
+    if participant1 == chain_state.our_address:
+        our_locksroot = locksroot_participant1
+        our_amount = amount_participant1
+        partner_locksroot = locksroot_participant2
+        partner_amount = amount_participant2
+    elif participant2 == chain_state.our_address:
+        our_locksroot = locksroot_participant2
+        our_amount = amount_participant2
+        partner_locksroot = locksroot_participant1
+        partner_amount = amount_participant1
+    else:
+        log.debug("Discarding settle event, we're not part of it")
+        # XXX this shouldn't happen because of the channel_state check
```

Comment:
Then we should log some more variables if we ever get to this.

Reply:
Then this should either be a `RaidenUnrecoverableError` or an assert.
Things that can't happen must not happen.